lib/cache/src/filesystem.rs

```diff
@@ -1,3 +1,4 @@
+#![cfg_attr(not(feature = "sys"), allow(unused))]
```

Comment:
Rather than this requiring the `sys` feature, can we create a new feature called `filesystem` that is enabled by default?

Why this? Because some targets like wasm32-wasi (which are not `sys`) could still make use of this without depending on sys.